src/layers/normalise.jl

```diff
@@ -10,7 +10,7 @@ _dropout_shape(s, dims) = tuple((i ∉ dims ? 1 : si for (i, si) ∈ enumerate(s
 _dropout_kernel(y::T, p, q) where {T} = y > p ? T(1 / q) : T(0)
 
 """
-    dropout([rng = rng_from_array(x)], x, p; dims=:, active=true)
+    dropout([rng = _rng_from_array(x)], x, p; dims=:, active=true)
```

Comment:
Maybe this shouldn't mention this internal function? If you aren't looking at the source, you don't need to know it exists:
```suggestion
    dropout([rng], x, p; dims=:, active=true)
```
Alternatively, if it's a function you may need to know about & overload for your `x`, then it should be left as `rng_from_array` and included in the manual.

Reply:
Interesting. The docstring does redirect users to the `Dropout` construct, so should we be concerned with users using this function?

If users are indeed using the `dropout` function, then it makes sense to treat `rng_from_array` as a public function and add it to the manual.

Reply:
The majority of docstrings use the following notation for specifying the default value as `rng_from_array`-
```
[rng=GLOBAL_RNG]
``` 
Example -
```
    kaiming_uniform([rng=GLOBAL_RNG], size...; gain = √2) -> Array
    kaiming_uniform([rng]; kw...) -> Function
```
```
    glorot_normal([rng=GLOBAL_RNG], size...; gain = 1) -> Array
    glorot_normal([rng]; kw...) -> Function
```
etc.

I think these should also be changed. Now that I think of it, maybe we should keep it as `rng_from_array` and add it to the manual.

Reply:
One point against mentioning `rng_from_array` in the docstring is that these functions don't actually use it! For most cases users will encounter it will behave as if it were, but there are enough edge cases that I'd prefer not to unintentionally mislead about expected behaviour.

Reply:
Ah, so should I keep it private, and update the docstring as `[rng]`?

Reply:
I stand corrected, it seems the functions in question were updated very recently to use it. Don't see much hurt in documenting `rng_from_array` and just noting it's an internal function, but you may want to wait for others to chime in. Another idea would be to call the zero-arg `rng_from_array` something else and document that. The current name is a bit of a misnomer anyways because there is no array involved in that method...

Reply:
Oh yes, something like `default_rng_value()`? Then maybe add both of them to the manual?